packages/components/src/button/index.js

```diff
@@ -44,19 +58,56 @@ export function Button( props, ref ) {
 		'is-busy': isBusy,
 		'is-link': isLink,
 		'is-destructive': isDestructive,
+		'has-text': !! icon && !! children,
+		// Ideally should be has-icon but this is named this way for BC
+		'components-icon-button': !! icon,
 	} );
 
-	const tag = href !== undefined && ! disabled ? 'a' : 'button';
-	const tagProps = tag === 'a' ?
+	const Tag = href !== undefined && ! disabled ? 'a' : 'button';
+	const tagProps = Tag === 'a' ?
 		{ href, target } :
 		{ type: 'button', disabled, 'aria-pressed': isPressed };
 
-	return createElement( tag, {
-		...tagProps,
-		...additionalProps,
-		className: classes,
-		ref,
-	} );
+	// Should show the tooltip if...
+	const showTooltip = ! additionalProps.disabled && (
```

Comment:
Disabled is already listed as a prop for the component:
```suggestion
	const showTooltip = ! disabled && (
```